### ISSUE TITLE
Transform network doubt/issue

### ISSUE THREAD

**Issue by dhorka** [fxia22/pointnet.pytorch#14]
Hello,

I have a doubt of the network STN3D, in the paper the authors says that after the max pooling there are only two layers and the output matrix is initialized as identity matrix. But I am not able to see this implementation on the code.  In the file [https://github.com/fxia22/pointnet.pytorch/blob/master/pointnet.py](pointnet.py) in the class stn3d what I can see is a third full connected layer that is not described on the paper but I am guessing that this stage is used in order to generate the output  matrix of 3x3 (that I think this is a correct) but my issue is related to this lines:

```
 x = self.fc3(x)

        iden = Variable(torch.from_numpy(np.array([1,0,0,0,1,0,0,0,1]).astype(np.float32))).view(1,9).repeat(batchsize,1)
        if x.is_cuda:
            iden = iden.cuda()
        x = x + iden
        x = x.view(-1, 3, 3)
        return x
```
Here, I am not able too see the init of this layer as a identity matrix. That I see is that in every iteration you are adding to the matrix calculated by the network an identity matrix but this is not the same that is described on the paper, right?

Another doubt that I have is related to the convolutional layers, why are you using 1d convolutional layers? In the tensorflow implementation the convolutional layers are 2d. Is it for any specific reason?

Thanks in advance

**Comment by fxia22** [comment]
Hi @dhorka , adding the variable here is equivalent to adding bias in the original implementation [here](https://github.com/charlesq34/pointnet/blob/master/models/transform_nets.py#L47).

**Comment by lyakaap** [comment]
@dhorka MLP can be implemented as Conv1D straightforwardly, but in Conv1D, we cannot gain cudnn optimization advantage. So I guess the reason why the author uses Conv2D instead of 1D is that.

**Comment by zeal-github** [comment]
@lyakaap Hi, do you mean that only conv2d layer can speed up using GPU? I am confused why the implementation of pointnet/++ are all using conv layer rather than fc layer straightly. Maybe because the cudnn?
Can you give me some manual about this knowledge? Thanks!

**Comment by lyakaap** [comment]
Yes, you're right.
